nemo/collections/common/data/lhotse/nemo_adapters.py

```diff
@@ -39,24 +39,29 @@ class LazyNeMoIterator(ImitatesDict):
     - "text" (overridable with ``text_field`` argument)
 
     Specially supported keys are:
+    - [recommended] "sampling_rate" allows us to provide a valid Lhotse ``Recording`` object without checking the audio file
```

Comment:
Hmm sampling rate in manifest row item? Wont it be loads of repetition?

Reply:
Depending on your setup it might, but it doesn't really make a difference if you don't need to pre-load the manifest into memory. Note that it is still an optional key -- just recommended if you want to avoid performing file header reads during iteration.

For some context - this has been my #1 struggle in maintaining NeMo format adaptation to Lhotse. Lhotse requires sampling rate to be known for each utterance because many transforms are applied lazily on the metadata level. If you know the duration, sampling rate, number of samples, etc. you can attach transforms on the meta-data level and resolve them lazily during actual audio loading operation.

 E.g. the following is valid lhotse code that'd apply a bunch of augmentations `cut = cut.resample(8000).perturb_speed(0.9).mix(noise_cut).reverb_rir().resample(16000)` -- no actual operation is performed until you call `cut.load_audio()`, but you can query the cut about `cut.duration`, `cut.num_samples`, `cut.sampling_rate`, and get reliable info.

Reply:
Why is it not possible to read the global cfg.sample_rate into Lhotse ?

Reply:
Lhotse has no concept of "dataset-level" anything. Everything is an item in a sequence with its own set of properties.